### PR TITLE
For loops changed to parallel int streams in Gandalf and Matras scraper

### PR DESCRIPTION
**Pull request by patkkka**

### CODE REVIEW

File: gandalfscraper/src/main/java/com/blackbook/gandalfscraper/scraper/Scraper.java [hunk 1]
```diff
@@ -36,16 +37,19 @@ public Scraper(WebConnector webConnector, LastPageChecker lastPageChecker) {
...
-            Document pageDoc = webConnector.connect(discountUrl + i);
-            bookDiscountData.addAll(extractBookElementsFromSinglePage(pageDoc));
-        }
+        IntStream.rangeClosed(1, lastPageNo)
+                .parallel()
+                .forEach(i -> {
+                    Document pageDoc = webConnector.connect(discountUrl + i);
+                    bookDiscountData.addAll(extractBookElementsFromSinglePage(pageDoc));
+                });
         return bookDiscountData;
```
Comment (Dzionek95) [inline gandalfscraper/src/main/java/com/blackbook/gandalfscraper/scraper/Scraper.java:46]: :+1:

File: gandalfscraper/src/main/java/com/blackbook/gandalfscraper/scraper/Scraper.java [hunk 2]
```diff
@@ -36,16 +37,19 @@ public Scraper(WebConnector webConnector, LastPageChecker lastPageChecker) {
...
     private List<BookDiscountData> extractBookElements(int lastPageNo) {
         List<BookDiscountData> bookDiscountData = new LinkedList<>();
-        for (int i = 0; i < lastPageNo; i++) {
-            Document pageDoc = webConnector.connect(discountUrl + i);
-            bookDiscountData.addAll(extractBookElementsFromSinglePage(pageDoc));
-        }
+        IntStream.rangeClosed(1, lastPageNo)
+                .parallel()
+                .forEach(i -> {
+                    Document pageDoc = webConnector.connect(discountUrl + i);
```
Comment (agawda) [inline gandalfscraper/src/main/java/com/blackbook/gandalfscraper/scraper/Scraper.java:43]: 💙

File: matrasscraper/src/main/java/com/blackbook/matrasscraper/scraper/Scraper.java [hunk 3]
```diff
@@ -45,6 +60,7 @@ private int extractLastPageNo(Document mainPageDoc) {
     private List<BookDiscountData> extractBookElementsFromSinglePage(Document document) {
         Elements bookElements = document.select("main .s-item.eqh");
         return bookElements.stream()
+                .parallel()
```
Comment (agawda) [inline matrasscraper/src/main/java/com/blackbook/matrasscraper/scraper/Scraper.java:63]: Oooh. 🖤

**Review by Dzionek95 — APPROVED** [review]
Cool!

**Review by agawda — APPROVED** [review]

**Pull request merged**